### DISCUSSION TITLE
O-RAN NearRT-RIC and xAPP Tutorial

### DISCUSSION THREAD

**Discussion by doogie05** [srsran/srsRAN_Project#762] [General Help] [answered]
Hi,
I am following the step by step instructions posted in O-RAN NearRT-RIC and xAPP Tutorial.
I completed required configurations, going through Running the Network section.

I am running the network in the order specified by the tutorial.
1.  Open5GS (using docker compose)
2.  NearRT-RIC (using flexRIC, ran it and matched the flexRIC console log output as shown in the tutorial)
3. gNB (ran it and matched the gNB console log output as shown in the tutorial)
4. UE

When I ran ue, I am getting PDU session error. (image below).
![image](https://github.com/user-attachments/assets/31a4deaf-393d-4ae3-a132-3fec42554a64)

I did enable ip forwarding and set iptable rules.
![image](https://github.com/user-attachments/assets/7ef1800c-f370-4fa0-890f-7d9c17f83b43)

my routing table configuration is as below, and I tied to ping google DNS per troubleshooting UE Issues.
I still get the network unreachable error:
![image](https://github.com/user-attachments/assets/0a9425df-2050-4953-9145-c2afac73b88d)

I rebooted my PC a couple of times and tried the same steps but I am stuck at the PDU session error.
Please comment.

Thanks.
James

**Comment by aithamid** [comment]
Try `sudo ip netns exec ue1 ip route add default via 10.45.1.1 dev tun_srsue` or try to do POSTROUTING on `ogstun`

**Reply by doogie05** [reply]
When I ran the command you suggested above, I am getting this error:
cannot find device "tun_srsue"

is tun_srsue device created when ./srsue ue_zmq.conf is called?
When i run srsue command, it is faulting out due to PDU Session error.

**Comment by pgawlowicz** [comment]
@doogie05 could you share gnb and srsUE logs?

**Reply by doogie05** [reply]
This is 5G Core Output: (NOTE: I am getting error message, 8805 failed):
![image](https://github.com/user-attachments/assets/d074ec74-2383-4c07-9ace-9a1ec8a31572)

gnb output:
![image](https://github.com/user-attachments/assets/85c53150-b318-4ed5-b9c4-6e5baf40bcde)

srsUE output;
![image](https://github.com/user-attachments/assets/c24843ef-5325-499e-a077-70c837e161e0)

**Reply by pgawlowicz** [reply, marked as answer]
It seems that SMF service in open5gs is not running, could you remove the container a build again?

**Reply by doogie05** [reply]
You are correct.  I rebuilt containers by `docker compose up --build 5gc`.
I can ping both ways now.
Thank you very much for your help.

**Comment by yagoda** [comment]
Hi @doogie05, 

In preparation for our upcoming workshop in DC, we are contacting some users of our E2 solution. Can you please share a few lines about your experience with the E2 interface, what you were trying to achieve, and how it worked out for you?

Kind Regards,
Justin
